snippet/snippet.go

```diff
@@ -30,45 +30,53 @@ func (snippets *Snippets) Load(includeDirs bool) error {
 	// Create a list of snippet files to load snippets from
 	var snippetFiles []string
 
-	// Load snippets from the main snippet file
-	// Raise an error if the file is not found / not configured
-	snippetFile := config.Conf.General.SnippetFile
-	if snippetFile != "" {
+	snippetFile, err := config.ExpandPath(config.Conf.General.SnippetFile)
+	if err == nil {
 		if _, err := os.Stat(snippetFile); err == nil {
-			snippetFiles = append(snippetFiles, snippetFile)
+			snippetFiles = append(snippetFiles, config.Conf.General.SnippetFile)
 		} else if !os.IsNotExist(err) {
 			return fmt.Errorf("failed to load snippet file. %v", err)
 		} else {
 			return fmt.Errorf(
 				`snippet file not found. %s
 Please run 'pet configure' and provide a correct file path, or remove this
 if you only want to provide snippetdirs instead`,
-				snippetFile,
+				config.Conf.General.SnippetFile,
 			)
 		}
 	}
 
-	// Optionally load snippets from snippet directories
 	if includeDirs {
-		for _, dir := range config.Conf.General.SnippetDirs {
+		for _, snippetDir := range config.Conf.General.SnippetDirs {
+			dir, err := config.ExpandPath(snippetDir)
+			if err != nil {
+				return fmt.Errorf("snippet directory not found. %s", snippetDir)
+			}
+
 			if _, err := os.Stat(dir); err != nil {
 				if os.IsNotExist(err) {
-					return fmt.Errorf("snippet directory not found. %s", dir)
+					return fmt.Errorf("snippet directory not found. %s", snippetDir)
 				}
-				return fmt.Errorf("failed to load snippet directory. %v", err)
+				snippetFiles = append(snippetFiles, getFiles(dir)...)
```

Comment:
Why add this here? It is triggered right below

Reply:
Must be an error on my part while resolving conflicts. I'll have it removed